.github/workflows/publish-pypi.yml

```diff
@@ -0,0 +1,280 @@
+    runs-on: ${{ matrix.platform.runner }}
+    strategy:
+      matrix:
+        platform:
+          - runner: windows-latest
+            target: x64
+          - runner: windows-latest
+            target: x86
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+          architecture: ${{ matrix.platform.target }}
+      - name: Download fable-library-py
+        uses: actions/download-artifact@v4
+        with:
+          name: fable-library-py
+          path: ./temp/fable-library-py
+      - name: Build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.platform.target }}
+          args: --release --out dist --find-interpreter
+          sccache: ${{ !startsWith(github.ref, 'refs/tags/') }}
+          working-directory: ./temp/fable-library-py
+      - name: Upload wheels
+        uses: actions/upload-artifact@v4
+        with:
+          name: wheels-windows-${{ matrix.platform.target }}
+          path: ./temp/fable-library-py/dist
+
+  macos:
+    needs: build-fable-library
+    runs-on: ${{ matrix.platform.runner }}
+    strategy:
+      matrix:
+        platform:
+          - runner: macos-13
+            target: x86_64
+          - runner: macos-14
+            target: aarch64
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+      - name: Download fable-library-py
+        uses: actions/download-artifact@v4
+        with:
+          name: fable-library-py
+          path: ./temp/fable-library-py
+      - name: Build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.platform.target }}
+          args: --release --out dist --find-interpreter
+          sccache: ${{ !startsWith(github.ref, 'refs/tags/') }}
+          working-directory: ./temp/fable-library-py
+      - name: Upload wheels
+        uses: actions/upload-artifact@v4
+        with:
+          name: wheels-macos-${{ matrix.platform.target }}
+          path: ./temp/fable-library-py/dist
+
+  sdist:
+    needs: build-fable-library
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download fable-library-py
+        uses: actions/download-artifact@v4
+        with:
+          name: fable-library-py
+          path: ./temp/fable-library-py
+      - name: Build sdist
+        uses: PyO3/maturin-action@v1
+        with:
+          command: sdist
+          args: --out dist
+          working-directory: ./temp/fable-library-py
+      - name: Upload sdist
+        uses: actions/upload-artifact@v4
+        with:
+          name: wheels-sdist
+          path: ./temp/fable-library-py/dist
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'release' || github.event_name == 'workflow_dispatch' }}
+    needs: [linux, musllinux, windows, macos, sdist]
```

Comment:
If I understand correctly, we use Github CI to build the binaries and then publish?

How do we trigger the release workflow? 

How is the version number determined? Is it based on the Github release version?

Reply:
- Yes, we use Github Ci to build the binaries for every platform and architecture
- The idea is that the release is triggered by a new Fable release e.g [5.0.0-alpha.12](https://github.com/fable-compiler/Fable/releases/tag/5.0.0-alpha.12). So we build a new fable-library for every Fable release. 
- The version is generated from the tag of the release so 5.0.0-alpha-12 becomes fable-library-5.0.0a12 (because of [PEP-440](https://peps.python.org/pep-0440/))